### PR TITLE
endpoints for blocking stations

### LINKED ISSUE #33: Create endpoint for blocking/unblocking stations

### PR DESCRIPTION
**Pull request by ArturChmura**
Closes #33

### CODE REVIEW

File: tests/test_stationsBlockedDetailViews.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    @pytest.fixture
+    def factory(self):
+        return APIRequestFactory()
+
+    def test_delete_station_blocked_detail_user_status(self, factory, station_blocked, user):
+        request = factory.delete(f'/api/stations/blocked/{station_blocked.pk}')
+        request.headers = {'Authorization': f'Bearer {user.user.username}'}
+
+        response = stations_blocked_detail(request, station_blocked.pk)
+        assert response.status_code == 403
```
Comment (debskipawel) [inline tests/test_stationsBlockedDetailViews.py:53]: Please change the asserted status codes in tests to 'status' enum to maintain consistency with Maciek's changes

File: tests/test_stationsBlockedViews.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    def test_post_stations_blocked_response(self, factory, admin, station_working):
+        body = json.dumps({'id': station_working.pk})
+        request = factory.post('/api/stations/blocked', content_type = 'application/json', data = body)
+        headers = {'Authorization': f'Bearer {admin.user.username}'}
+        headers.update(request.headers)
+        request.headers = headers
+
+        response = stations_blocked(request)
+        assert response.status_code == 201
+        assert json.loads(response.content) == {
```
Comment (debskipawel) [inline tests/test_stationsBlockedViews.py:135]: I don't think 2 asserts in one tests are correct, it should be two separate tests

**Review by debskipawel — CHANGES_REQUESTED** [review]

File: tests/test_stationsBlockedDetailViews.py (hunk 1 above)
Comment (ArturChmura) [reply tests/test_stationsBlockedDetailViews.py]: done

File: tests/test_stationsBlockedViews.py (hunk 2 above)
Comment (ArturChmura) [reply tests/test_stationsBlockedViews.py:135]: done

File: BikeRentalApi/endpoints/stationsBlocked.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    stream = io.BytesIO(request.body)
+    data = JSONParser().parse(stream)
+
+    try:
+        id = int(data['id'])
+    except KeyError:
+        return JsonResponse({"message": "Bad data"}, status = status.HTTP_400_BAD_REQUEST)
+
+    station = BikeStation.objects.filter(id = id).first()
```
Comment (gardzielb) [inline BikeRentalApi/endpoints/stationsBlocked.py:33]: For single object better to use `BikeStation.objects.get()` than `filter`

File: BikeRentalApi/endpoints/stationsDetailsBlocked.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from rest_framework import status
+
+from BikeRentalApi.enums import Role, StationState
+from BikeRentalApi.models import BikeStation
+
+
+def delete(user, pk):
+    if user.role != Role.Admin:
+        return JsonResponse({"message": "Forbidden"}, status = status.HTTP_403_FORBIDDEN)
+    station = BikeStation.objects.filter(pk = pk).first()
```
Comment (gardzielb) [inline BikeRentalApi/endpoints/stationsDetailsBlocked.py:11]: For single object better to use `BikeStation.objects.get()` than `filter`

**Review by gardzielb — CHANGES_REQUESTED** [review]
Replace `objects.filter()` with `objects.get()` in two places. The rest seems nice.

File: BikeRentalApi/endpoints/stationsBlocked.py (hunk 3 above)
Comment (ArturChmura) [reply BikeRentalApi/endpoints/stationsBlocked.py]: done

File: BikeRentalApi/endpoints/stationsDetailsBlocked.py (hunk 4 above)
Comment (ArturChmura) [reply BikeRentalApi/endpoints/stationsDetailsBlocked.py]: done

**Review by gardzielb — APPROVED** [review]
Well done

**Pull request merged**